packages/uni_app/lib/view/restaurant/restaurant_page_view.dart

```diff
@@ -257,22 +263,61 @@ class _RestaurantPageViewState extends GeneralPageViewState<RestaurantPageView>
     final menuItems =
         getRestaurantMenuItems(dayOfWeek, restaurant, locale) ?? [];
     return menuItems.isNotEmpty
-        ? RestaurantCard(
-            name: RestaurantUtils.getRestaurantName(
-              context,
-              locale,
-              restaurant.namePt,
-              restaurant.namePt,
-              restaurant.period,
-            ),
-            icon: RestaurantUtils.getIcon(
-              restaurant.typeEn ?? restaurant.typePt,
+        ? GestureDetector(
+            onTap: () {
+              if (restaurant.openingHours.isNotEmpty) {
+                showDialog<ModalDialog>(
+                  context: context,
+                  builder: (context) {
+                    return ModalDialog(
+                      children: [
+                        ModalServiceInfo(
+                          name: restaurant.namePt,
+                          durations: restaurant.openingHours
+                            ..sort((a, b) => a.compareTo(b)),
+                        ),
+                        if (restaurant.email != '')
+                          ModalInfoRow(
+                            title: S.of(context).email,
+                            description: restaurant.email,
+                            onPressed: () => launchUrlWithToast(
+                              context,
+                              'mailto:${restaurant.email}',
+                            ),
+                            icon: UniIcon(
+                              UniIcons.paperPlaneTilt,
+                              color: Theme.of(context).primaryColor,
+                            ),
+                            optionalIcon: const UniIcon(UniIcons.caretRight),
+                          ),
+                      ],
+                    );
+                  },
+                );
+              } else {
+                ToastMessage.warning(
+                  context,
+                  S.of(context).no_info,
+                );
```

Comment:
Personally, I don't like it. It should have no alert. No feedback is feedback enough.